src/containers/barray.mli

```diff
@@ -64,10 +64,10 @@ val fill : 'a array -> int -> int -> 'a -> unit
     Raise [Invalid_argument "Array.fill"] if [ofs] and [len] do not
     designate a valid subarray of [a]. *)
 
-val to_list : 'a array -> 'a list
+val toList : 'a array -> 'a list
 (** [Array.to_list a] returns the list of all the elements of [a]. *)
 
-val of_list : 'a list -> 'a array
+val fromList : 'a list -> 'a array
```

Comment:
The issue is that you can't make this efficient at scale. You either have to do an O(N) scan of the list to get its count, or you have to do multiple array copies as you discover the need for a bigger array. Either option is inefficient. The only reason OCaml developers do this is that they don't have better datastructures available. Immutable.re solves this problem by providing Vector, Deque and Stack, all that have O(1) count operations.